examples/demo_bgp.pp

```diff
@@ -23,7 +23,8 @@
     ensure                                 => present,
 
     router_id                              => '192.168.0.66',
-    route_distinguisher                    => 'auto',
+    # route_distinguisher is not supported on all platforms
```

Comment:
Add platform selector instead of commenting this out completely.

Here is an example: https://github.com/cisco/cisco-network-puppet-module/blob/develop/examples/demo_portchannel.pp#L24